pkg/client/cluster_config.go

```diff
@@ -26,19 +26,21 @@ import (
 
 // NewClusterConfig wraps an existing config's roundtripper
 // with a custom cluster aware roundtripper.
+//
+// Note: it is the caller responsibility to make a copy of the rest config
 func NewClusterConfig(cfg *rest.Config) *rest.Config {
```

Comment:
I'm tempted to deprecate this function name (or just delete it entirely) and replace it with something like `func EnableMultiCluster(cfg *rest.Config)`.

If we proceed with this, and it's mutating the input, we can remove the `return`.

Reply:
I'd prefer to leave the return statement. It allows for chaining, i.e. `rest.AddUserAgent(kcpclienthelper.EnableMultiCluster(cfg))`

wdyt?

Reply:
Ah, didn't think about chaining. That's fine!